netbox_branching/signal_receivers.py

```diff
@@ -136,6 +189,46 @@ def handle_branch_event(event_type, branch, user=None, **kwargs):
 signals.post_revert.connect(partial(handle_branch_event, event_type=BRANCH_REVERTED))
 
 
+@receiver(pre_delete)
+def validate_object_deletion_in_branch(sender, instance, **kwargs):
+    """
+    Validate that objects being deleted in a branch still exist in main.
+    """
+    # Skip Branch objects - they have their own validation
+    if sender == Branch:
+        return
+
+    # Only validate if we're in a branch
+    branch = active_branch.get()
+    if branch is None:
+        return
+
+    # Check if this model supports branching
+    try:
+        object_type = ObjectType.objects.get(app_label=instance._meta.app_label, model=instance._meta.model_name)
+        if 'branching' not in object_type.features:
+            return
+    except ObjectType.DoesNotExist:
+        return
+
+    # For deletions, check if the object exists in main
+    if hasattr(instance, 'pk') and instance.pk is not None:
+        model = instance.__class__
+        try:
+            # Try to get the object from main database (explicitly use default connection)
+            model.objects.using(DEFAULT_DB_ALIAS).get(pk=instance.pk)
```

Comment:
It seems like we're doing this checking in `main` differently in this signal receiver than in `validate_branching_operations()`. Is that intentional? If not, then I think we should stick with way or the other to be consistent.